src/react/site/home/index.tsx

```diff
@@ -1,25 +1,54 @@
-import React from "react";
+import React, { useEffect } from "react";
 import clsx from "clsx";
 import Threads from "../components/threads";
 import { getIconClassBySlug } from "../utils/forums";
 import { useStore } from "../../../utils/store";
-import { getUsername } from "../../../injected/utils/data";
-import { Thread } from "../../../domains/thread";
+import {
+  getFavorites,
+  getForumLastThreads,
+  getLastNews,
+  getUsername,
+  getUserLastPosts,
+} from "../../../injected/utils/data";
+import { useQuery } from "@tanstack/react-query";
 
-function Home({
-  favorites,
-  lastThreads,
-  lastNews,
-  userLastPosts,
-}: {
-  favorites: Thread[];
-  forums: { name: string; url: string; slug: string }[];
-  lastThreads: Thread[];
-  lastNews: Thread[];
-  userLastPosts: Thread[];
-}) {
+function Home({ onLoad }: { onLoad: () => void }) {
   const { forumsLastVisited } = useStore();
 
+  const { data: lastThreads, isPending: lastThreadsPending } = useQuery({
+    queryKey: ["lastThreads"],
+    queryFn: () => getForumLastThreads(),
+  });
+
+  const { data: userLastPosts, isPending: userLastPostsPending } = useQuery({
+    queryKey: ["userLastPosts"],
+    queryFn: () => getUserLastPosts(getUsername()),
+  });
+
+  const { data: favorites, isPending: favoritesPending } = useQuery({
+    queryKey: ["favorites"],
+    queryFn: () => getFavorites(),
+  });
+
+  const { data: lastNews, isPending: lastNewsPending } = useQuery({
+    queryKey: ["lastNews"],
+    queryFn: () => getLastNews(),
+  });
+
+  useEffect(() => {
+    [
+      lastThreadsPending,
+      userLastPostsPending,
+      favoritesPending,
+      lastNewsPending,
+    ].every((isPending) => !isPending) && onLoad();
+  }, [
+    lastThreadsPending,
+    userLastPostsPending,
+    favoritesPending,
+    lastNewsPending,
+  ]);
```

Comment:
Why is this useEffect needed?

Reply:
It's not absolutely needed. It just waits for the all data to be ready before showing the content on the first ever page load (before the storage has been populated at least once).

Without it, we'd need to define the min-heights that I removed in [37296df](https://github.com/jvidalv/mv-ignited/pull/3/commits/37296dfe527aef1449b87508c267309c94ff3af3) and the contents would appear individually as they are ready. I feel a bit lazy to record a demo 😅 

It's kind of a stop-gap to avoid showing a loading spinner, or some sort of skeleton components. I think it's something that could be better handled separately.

But in short, I think the `useEffect` could be removed and impact would be minimal.

Reply:
Got it, it leverages my ""solution"" (hacky) showContent and triggers it when all of them are populated. 

Yeah for now its good, for future, I would advocate on removing this show/hideContent stuff I did and instead leverage fully react and skeletons.